docs/configuration/secure.md

```diff
@@ -0,0 +1,571 @@
+
+Apache2:
+
+    <FilesMatch "\.(sec)$">
+        Order Deny,Allow
+        Deny from All
+    </FilesMatch>
+
+Nginx:
+
+    location ~ [^/]\.sec {
+        deny all;
+    }
+
+### Prevent execution of PHP-Code in data-directory
+There may be situations where there is no oppurtunity to disallow uploading php-files e.g. in Computer Science courses. In this case you SHOULD disallow these uploaded code to be executed by the webserver.
+
+Apache2:
+```
+    <LocationMatch "/data/.*(?i)\.(php)$">
+        Order Deny,Allow
+        Deny from All
+    </LocationMatch>
+```
+Nginx:
+```
+    location ~* /data/.*\.php {
+        deny all;
+    }
+```
+
+### Deny Access to local Git-Directory
+If you installed ILIAS via git, access the local Git-Directory (.git) SHOULD be denied for visitors via web.
+
+Apache2:
+```
+    <Directorymatch "^/.*/\.git/">
+        Order deny,allow
+        Deny from all
+    </Directorymatch>
+```
+Nginx:
+```
+    location /.git {
+        deny all;
+    }
+```
+
+
+
+## Integrity check of ILIAS code in docroot
+Local changes of the code of ILIAS can indicate a potential intrusion.
+
+To determine local changes of the code of ILIAS use `git status` / `git diff`.
+This will show you the uncommited local changes. 
+(Beware: Committed local changes remain undetected using this method.)
+(If you have conscious code local changes, this can lead to a false positive.)
+
+## Use WebAccessChecker
+In previous versions of ILIAS, it might have been possible to access SCORM, Media Files and User Profile Images without beeing logged in by guessing the proper URL and no measures were taken by the admin to deny such access. Since ILIAS 5.1, a new WebAccessChecker (WAC) is activated by default. To make use of WAC you MUST enable ``mod_rewrite`` in your Apache configuration.
+
+Please note that this will not work with Nginx as ``.htaccess-files`` are not supported. Instead you MUST add the following to your Nginx configuration file (please note that running ILIAS with Nginx isn't officially supported and certain features like Shibboleth won't work):
+
+This is a NGINX recommended configuration. (note: inside the `%docroot_dir%/data` no PHP will be proceed)
+
+
+```
+    server {
+        [...]
+        location ~ /data/ {
+            set $root $document_root;
+            rewrite ^/data/(.*)/(.*)/(.*)$ /Services/WebAccessChecker/wac.php last;
+
+            location ~ [^/]\.php(/|$) { access_log off; log_not_found off; deny all; }
+        }
+
+        location ^~ /secured-data {
+            # Protected, only working for subrequests (X-Accel-Redirect)
+            alias $root/data;
+            internal;
+
+            location ~ [^/]\.php(/|$) { access_log off; log_not_found off; deny all; }
+        }
+         [...]
+    }
+```
+
+
+** [ilFileDelivery](https://github.com/ILIAS-eLearning/ILIAS/blob/release_5-4/Services/FileDelivery/classes/override.php.template)** (concerns NGINX/PHP-FPM):
```

Comment:
What is "ilFileDelivery" and why would I use it? Which implications, if any, might be noteworthy?

Reply:
ilFileDelivery is also described in the template files itself.

For a good performance, it is suggested.
OK, this is also more a performance improvement.